src/main/java/org/apache/sysds/runtime/instructions/fed/FEDInstructionUtils.java

```diff
@@ -220,8 +221,11 @@ else if(inst instanceof QuaternaryCPInstruction) {
 		}
 		else if(inst instanceof SpoofCPInstruction) {
 			SpoofCPInstruction instruction = (SpoofCPInstruction) inst;
-			if(instruction.getOperatorClass().getSuperclass() == SpoofCellwise.class && instruction.isFederated(ec))
+			if((instruction.getOperatorClass().getSuperclass() == SpoofCellwise.class
+					|| instruction.getOperatorClass().getSuperclass() == SpoofRowwise.class)
+				&& instruction.isFederated(ec)) {
```

Comment:
Do you cover cases where the federated data is split on columns?
If not then this place would be ideal to ad another verification.

Reply:
Column partitioned federated data is not supported within federated rowwise spoof instructions.
You're right, if we want to support column paritioned federated data as well (even without a federated instruction), we need to add a verification here, such that we don't replace the spoof instruction with a federated spoof instruction.
Thanks, I will add this :)